### PR TITLE
fix: checksum rustup in CI

### PR DESCRIPTION
**Pull request by jacobmarble**
Best to verify the checksum of anything fetched from the internet.

### CODE REVIEW

**Review by crepererum — APPROVED** [review]

**Review by domodwyer — APPROVED** [review]
I trust your hashes too.

File: scripts/install_rustup.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
```
Comment (alamb) [inline scripts/install_rustup.sh:2]:
```suggestion

# Use rustup to install rust tool chain after first
# comparing to a known checksum
```

**Comment by mkmik** [comment]
> I trust your hashes too.

I wish rustup served a .sha256 or a SHA256SUMS like in the good old days; so it would be very cheap to verify during review

File: scripts/install_rustup.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+set -eu -o pipefail
+
+RUSTUP_VERSION=1.24.3
+RUSTUP_HASH=a3cb081f88a6789d104518b30d4aa410009cd08c3822a1226991d6cf0442a0f8
+
+# Notice --location to follow redirects -- important on Ubuntu 20.04
+curl --proto '=https' --tlsv1.2 -sSf --location --max-redirs 1 \
```
Comment (mkmik) [inline scripts/install_rustup.sh:9]: I wish curl could just add this verify sha functionality built in.

Comment (mkmik) [reply scripts/install_rustup.sh:9]: fwiw, in the past I've been using this https://github.com/mkmik/runck

**Comment by NGA-TRAN** [comment]
Great to have this after the security training

**Comment by shepmaster** [comment]
> I wish rustup served a .sha256 or a SHA256SUMS 

Are the ones listed on https://rust-lang.github.io/rustup/installation/other.html what you are looking for?

**Comment by mkmik** [comment]
yeah, but if they were in a standard location they could be used by a tool like https://getsum.pub/

**Comment by mkmik** [comment]
anyway, not all transparent log checksum checkers require that file, for example https://github.com/transparencylog/tl doesn't. I architected getsum so that a random attacker cannot just squat a bunch of urls with some bad hashes preventing the legitimate owner of domains to use some filenames.

**Comment by mkmik** [comment]
> > I wish rustup served a .sha256 or a SHA256SUMS
> 
> Are the ones listed on https://rust-lang.github.io/rustup/installation/other.html what you are looking for?

@shepmaster the problem with those paths is that they don't encode the version name in the path, e.g.:

* https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init
* https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init.sha256

**Comment by carols10cents** [comment]
> @shepmaster the problem with those paths is that they don't encode the version number in the path, e.g.:
>
> *    static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init
> *   static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init.sha256

Further down on https://rust-lang.github.io/rustup/installation/other.html it says:

> You can fetch an older version from `https://static.rust-lang.org/rustup/archive/{rustup-version}/{target-triple}/rustup-init[.exe]`

**Comment by mkmik** [comment]
ah, but that's the sha for the binary, not for the shell script itself.

I.e. right now we're downloading https://raw.githubusercontent.com/rust-lang/rustup/1.24.3/rustup-init.sh and we'd want a reviewer to easily verify the checksum for that

**Pull request merged**